reactive.js

```diff
@@ -0,0 +1,56 @@
+/* reactive engine. the code here combines functions exposed by the
+ * library into a single function to be run. i keep all code depending
+ * on Bacon here, so if somebody will want to stop using Bacon in the
+ * future, it will be easier */
+var lib = require('./lib2');
+var Bacon = require('bacon');
+
+// this is an adapter function. it gets a Pouch changes object and it
+// returns a Bacon stream
+function changesToStream(changes) {
+  return Bacon.fromBinder(function(sink) {
```

Comment:
i still have to understand how errors are handled. intuitively, i would think that they get propagated to all streams derived from this one. so i think that i will try to add an error handler on the last stream, `outgoing`. i need to figure this out with some manual tests, though.

i am not sure how `fromEvent` works. `Bacon.fromEvent(changes, 'change')` would probably generate a stream of change events, but i think that it would never close the stream or intercept errors. Maybe we do not even care about closing the stream, but we care about errors for sure